File: lessons/compiled_lang/go/src/i10.subroutine.go
package main

import "fmt"

// package-level vars are directly visible and mutable from any function
//  in this package - no "global" keyword needed like Python.
var pond = 500
var captured = 0

func fish() {
	pond -= 150
	captured += 150
}

func main() {
	fmt.Printf("We have %d in this pond.\n", pond)

	fish()
	fmt.Printf("Fishing from the main pond... We now have %d in the main pond.\n", pond)

	fish()
	fmt.Printf("Fishing from the main pond... We now have %d in the main pond.\n", pond)

	fish()
	fmt.Printf("Fishing from the main pond... We now have %d in the main pond.\n", pond)

	fmt.Printf("We now have a total of %d fish captured\n", captured)
}

File: lessons/compiled_lang/cs/src/i10.subroutine.cs
using System;

class I10Subroutine
{
    // static fields are directly visible and mutable from any static
    //  method in this class - no "global" keyword needed like Python.
    static int pond = 500;
    static int captured = 0;

    static void Fish()
    {
        pond -= 150;
        captured += 150;
    }

    static void Main()
    {
        Console.WriteLine($"We have {pond} in this pond.");

        Fish();
        Console.WriteLine($"Fishing from the main pond... We now have {pond} in the main pond.");

        Fish();
        Console.WriteLine($"Fishing from the main pond... We now have {pond} in the main pond.");

        Fish();
        Console.WriteLine($"Fishing from the main pond... We now have {pond} in the main pond.");

        Console.WriteLine($"We now have a total of {captured} fish captured");
    }
}
